lib/global_uid/base.rb

```diff
@@ -164,10 +174,21 @@ def self.get_connections(options = {})
       with_connections {}
     end
 
+    def self.validate_increment!(new_record_id, connection)
+      server = self.servers.find { |s| connection.current_database.include?(s[:name]) }
+      server[:allocated_ids] << new_record_id
```

Comment:
It seems like the `server[:allocated_ids]` list will grow and grow over time. It may only be integers, but it seems like eventually performance will suffer, especially if we're doing a linear scan of the array every time we generate an ID. Perhaps we should keep only a window of the last 100 IDs generated, or even just the last ID?

Reply:
Spot on! This is still a draft PR, I've got some changes still I'd like to make and one of them was for this but I wanted the change paired with a performance spec to ensure we've got a benchmark and it never regresses in the future.